test/src/Test/Finance/Asset/Account.daml

```diff
@@ -0,0 +1,97 @@
+module Finance.Asset.Account where
```

Comment:
I'm particularly interested in: does the account behave as expected if we eg. have an empty nominee set? and do the signatories have the appropriate rights or are we limiting too much what they can do? we basically give all the rights to the nominees, but we need to think what the signatories authorization should let them do as well. or are we implicitly saying that the account provider should always also be a nominee for things to work as expected? Just some questions we should look to answer

Reply:
Very good questions - something that I've been trying to answer. In the current implementation as per this commit, I took the decision to separated the concerns of the signatories and the nominee's into : Signatories _back_ accounts whereas nominee's preform _actions_ on accounts. Note - only an account provider could be added as a nominee, not the account owner due to the `ensure` clause on the template. I need to spend more time on this as I'm not sure this is the correct approach when I sit back and think about it.